turbopack/crates/turbo-rcstr/src/lib.rs

```diff
@@ -44,13 +56,41 @@ use turbo_tasks_hash::{DeterministicHash, DeterministicHasher};
 // If you want to change the underlying string type to `Arc<str>`, please ensure that you profile
 // performance. The current implementation offers very cheap `String -> RcStr -> String`, meaning we
 // only pay for the allocation for `Arc` when we pass `format!("").into()` to a function.
-#[derive(Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
-#[serde(transparent)]
-pub struct RcStr(Arc<String>);
+pub struct RcStr {
+    unsafe_data: TaggedValue,
```

Comment:
This needs a `Drop` impl that attempts to turn `TaggedValue` back into an `Arc` if it has that tag and decrements the refcount.